app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb

```diff
@@ -128,12 +128,48 @@ def process_collection_item(item, key)
       new_result
     end
 
+    def cross_link_amazon_node(new_result)
+      _, az, uid = URI(new_result[:identity_infra]).path.split('/')
+      az_id = AvailabilityZone.find_by(:name => az).try(:id)
+      Vm.where(:uid_ems              => uid,
+               :availability_zone_id => az_id,
+               :type                 => ManageIQ::Providers::Amazon::CloudManager::Vm.name)
+    end
+
+    def cross_link_google_node(new_result)
+      # TODO
+    end
+
+    def cross_link_node(new_result)
+      # Establish a relationship between this node and the vm it is on (if it is in the system)
+      # supported relationships: oVirt, Openstack and VMware.
+      uuid_types = [ManageIQ::Providers::Redhat::InfraManager::Vm.name,
```

Comment:
@abonas 
These are types that are being searched and compared by their uuid, It was working before with this. I hope that it can also be compared with their providerID and then we can do all of them together but currently I don't have any of these providers so I tried not to change they way it works for them.
Can you suggest a better name?